File: Crassus.Native.iOS/ViewControllerDelegate.c
//
//  ViewControllerDelegate.c
//  TheObiG LLC
//
//  Created by SolomonRain on 2/15/16.
//  Copyright © 2016 Crassus. All rights reserved.
//

#include "ViewControllerDelegate.h"

void* CreateViewController(SingleObjectAction viewDidLoad, ObjectActionBlockObjectFunction function)
{
    return (*function)(viewDidLoad);
}

void* GetView(void* viewController, Exchanger function)
{
    return (*function)(viewController);
}

void ReleaseViewController(void* viewController, SingleObjectAction action)
{
    (*action)(viewController);
}

void* AutoReleaseViewController(void* viewController, Exchanger function)
{
    return (*function)(viewController);
}

void* RetainViewController(void* viewController, Exchanger function)
{
    return (*function)(viewController);
}

File: Crassus.Native.iOS/ViewControllerDelegate.m
//
//  ViewControllerDelegate.m
//  TheObiG LLC
//
//  Created by SolomonRain on 2/15/16.
//  Copyright © 2016 Crassus. All rights reserved.
//

@import UIKit;
#import "Crassus_Native_iOS-Swift.h"

void * CrassusCreateViewController(SingleObjectAction viewDidLoad)
{
    return (void *)CFBridgingRetain([CrassusViewController create:viewDidLoad]);
}

void * CrassusGetView(void * viewController)
{
    return (__bridge void *)((__bridge UIViewController*)viewController).view;
}